src/Microsoft.DotNet.Interactive.Http/HttpKernelExtension.cs

```diff
@@ -34,3 +35,23 @@ public class HttpKernelExtension
         }
     }
 }
+
+internal static class DateTimeOffsetExtensions
+{
+    public static DateTimeOffset AddOffset(this DateTimeOffset dateTime, int offset, string options)
+    {
+        return options switch
+        {
+            "ms" => dateTime.AddMilliseconds(offset),
+            "s" => dateTime.AddSeconds(offset),
+            "m" => dateTime.AddMinutes(offset),
+            "h" => dateTime.AddHours(offset),
+            "d" => dateTime.AddDays(offset),
+            "w" => dateTime.AddDays(offset * 7),
+            "M" => dateTime.AddMonths(offset),
+            "Q" => dateTime.AddMonths(offset * 3),
+            "y" => dateTime.AddYears(offset),
+            _ => throw new ArgumentException("Invalid offset option", nameof(options))
```

Comment:
minor: Consider renaming `options` to `offsetFormatString`.

```suggestion
            _ => throw new ArgumentException("Invalid offset format string", nameof(offsetFormatString))
```